src/control/xojfile/SaveHandler.cpp

```diff
@@ -333,11 +308,9 @@ void SaveHandler::saveTo(OutputStream* out, const fs::path& filepath, ProgressLi
     out->write("<?xml version=\"1.0\" standalone=\"no\"?>\n");
     root->writeOut(out, listener);
 
-    for (GList* l = this->backgroundImages; l != nullptr; l = l->next) {
-        auto* img = static_cast<BackgroundImage*>(l->data);
-
-        auto tmpfn = (fs::path(filepath) += ".") += img->getFilepath();
-        if (!gdk_pixbuf_save(img->getPixbuf(), tmpfn.u8string().c_str(), "png", nullptr, nullptr)) {
+    for (BackgroundImage& img: backgroundImages) {
```

Comment:
img is only used as const&:
```suggestion
    for (BackgroundImage const& img: backgroundImages) {
```